packages/common/index.ts

```diff
@@ -1392,6 +1397,51 @@ export type sendEmailAsync = {
   subject: string
   type: asyncQuestionEventType
 }
+
+export class CourseSettingsResponse {
+  @IsInt()
+  courseId!: number
+
+  @IsBoolean()
+  chatBotEnabled!: boolean
+
+  @IsBoolean()
+  asyncQueueEnabled!: boolean
+
+  @IsBoolean()
+  adsEnabled!: boolean
+
+  @IsBoolean()
+  queueEnabled!: boolean
+
+  @IsOptional()
+  @IsBoolean()
+  settingsFound?: boolean = true //this is mostly just for debugging purposes by viewing network responses
+
+  constructor(init?: Partial<CourseSettingsResponse>) {
+    Object.assign(this, init)
+  }
+}
+
+const validFeatures = [
```

Comment:
Why not using enum instead? Here is an example:

```
export enum Role {
  STUDENT = 'student',
  TA = 'ta',
  PROFESSOR = 'professor',
}
```

Reply:
I have thought about this and it's probably not worth the refactor.
Also, the only time validFeatures is used is right below it, and changing validFeatures to an enum kinda just makes the code look worse:

```tsx
export enum ValidFeatures {
  ChatBotEnabled = 'chatBotEnabled',
  AsyncQueueEnabled = 'asyncQueueEnabled',
  AdsEnabled = 'adsEnabled',
  QueueEnabled = 'queueEnabled',
}

export class CourseSettingsRequestBody {
  @IsBoolean()
  value!: boolean

  @IsEnum(ValidFeatures)
  feature!: ValidFeatures

  static isValidFeature(feature: string): boolean {
    return Object.values(ValidFeatures).includes(feature as ValidFeatures)
  }
}
```

vs

```tsx
const validFeatures = [
  'chatBotEnabled',
  'asyncQueueEnabled',
  'adsEnabled',
  'queueEnabled',
]

export class CourseSettingsRequestBody {
  @IsBoolean()
  value!: boolean

  @IsIn(validFeatures)
  feature!: string

  static isValidFeature(feature: string): boolean {
    return validFeatures.includes(feature)
  }
}
```